File: AccessChanger.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;
public class AccessChanger{
    public static String[] convert(String[] program){
        for(int i = 0; i < program.length; i++){
            String line = program[i];
            String code = line.replaceFirst("//.*", "");
            String comment = "";
            Pattern pattern = Pattern.compile("//.*");
            Matcher matcher = pattern.matcher(line);
            if(matcher.find())
                comment = matcher.group();
            program[i] = code.replaceAll("->", ".") + comment;
        }
        return program;
    }
}

File: AccessChanger.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
class AccessChanger{
    public:
    	vector<string> convert(vector <string> program);
};

vector<string> AccessChanger::convert(vector <string> program){
    for(int i = 0; i < program.size(); i++){
        string line = program[i];
		string processed_line = "";
		bool at_comment = false;
		
		int j = 0;
		while(j < line.length()){
			if(line[j] == '/' && line[j + 1] == '/'){
				at_comment = true;
				processed_line += line.substr(j, string::npos);
				break;
			}
			if(!at_comment){
				if(line[j] == '-' && line[j + 1] == '>'){
					processed_line += '.';
					j += 2;
				}else{
					processed_line += line[j];
					j++;
				}
			}
		}
		program[i] = processed_line;
    }
    return program;
}

int main(){
	AccessChanger a;
	vector<string> program;
	program.push_back("asdf->asdf");
	program.push_back("//comment");
	program.push_back("->->-->//comment");
	program = a.convert(program);
	for(int i = 0; i < program.size(); i++)
		cout<<program[i]<<endl;
	

	return 0;
}
